### PR TITLE
make sm120 matmul configurable and support all m/n/k-major combinations

### PR DESCRIPTION
**Pull request by yihuawei**
## Before submitting

- [x] I agree to license this contribution under the terms of [LICENSE.txt](https://github.com/NVIDIA/cudnn-frontend/blob/develop/LICENSE.txt).
- [x] I ran `pre-commit run` and committed any formatting changes.
- [x] I added GitHub labels: one `cat-*`, one or more `mod-*`, and one `orig-*` (see [label list](https://github.com/NVIDIA/cudnn-frontend/labels)).
- [x] I set the **Milestone** and **Projects** fields in the sidebar (required to merge; maintainers can set these for external contributions).

## Summary by CodeRabbit

* **New Features**
  * Expanded SM120 matrix multiplication support across configurable tile sizes, warp layouts, K widths, and operand memory layouts.
  * Added support for 8-bit operands with additional major-layout combinations and transposed loading.
  * Improved configuration validation and pipeline selection for supported geometries.

* **Bug Fixes**
  * Registry configurations without CTA group settings now use a safe default.
  * Improved memory-layout and swizzle validation across supported configurations.

* **Tests**
  * Expanded correctness and rendering coverage across SM120 configurations, layouts, geometries, and edge cases.

### CODE REVIEW

File: python/cudnn/gemm/frost/tile_config.py [hunk 1]
```diff
@@ -664,11 +673,27 @@ def _geom_sm120(
         cga_size_m=cga_size_m,
         cga_size_n=cga_size_n,
         cga_size_k=1,
-        warps_per_cta=_SM120_WARPS_PER_CTA,
+        warps_per_cta=ConfigSm120.DEFAULT_WARPS_PER_CTA,
         split_k_slices=1,
     )
 
 
+_SM120_SMEM_CAP_BYTES = 99 * 1024
```
Comment (yanqinz2) [inline python/cudnn/gemm/frost/tile_config.py]: Do not hardcode it. It should be queried from device

File: python/cudnn/gemm/frost/tile_config.py [hunk 2]
```diff
@@ -664,11 +673,27 @@ def _geom_sm120(
...
+
+
+def _sm120_smem_feasible(cta_m: int, cta_n: int, kb: int, num_compute_warps: int) -> bool:
+    per_stage = (cta_m + cta_n) * kb + 16
+    staging = 4 * _SM120_STG_STAGE_ELEMS * num_compute_warps
+    stages = min((_SM120_SMEM_CAP_BYTES - _SM120_SMEM_FIXED_RESERVE) // per_stage, 16)
+    return stages - -(-staging // per_stage) >= 1
+
+
+def _sm120_reg_feasible(cta_m: int, cta_n: int, num_compute_warps: int) -> bool:
```
Comment (yanqinz2) [inline python/cudnn/gemm/frost/tile_config.py]: Could you explain what does this part do?

File: benchmark/gemm/frost/benchmark_matmul_sm120.py [hunk 3]
```diff
@@ -1,342 +1,238 @@
 # SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (yanqinz2) [inline benchmark/gemm/frost/benchmark_matmul_sm120.py]: Could you incorporate this benchmark into benchmark_matmul? You can either add a sm120 flag or something else. Just not to make a standalone file for benchmarking/testing script for a specific pipeline

File: python/cudnn/gemm/frost/kernel_templates/sm120_matmul.py [hunk 4]
```diff
@@ -101,17 +112,30 @@
 
 # ---- v1 scope guards (fail at render/import, not at runtime) ---------------
```
Comment (yanqinz2) [inline python/cudnn/gemm/frost/kernel_templates/sm120_matmul.py]: Why we need assert in the kernel template? These things need to be done in the compiler

File: python/cudnn/gemm/frost/tile_config.py (hunk 2 above)
Comment (yihuawei) [reply python/cudnn/gemm/frost/tile_config.py]: On SM120, all accumulator results are stored in registers. If the tile size specified in the configuration is too large, it can cause register spilling. This check is intended to prevent that.

File: python/cudnn/gemm/frost/kernel_templates/sm120_matmul.py (hunk 4 above)
Comment (yihuawei) [reply python/cudnn/gemm/frost/kernel_templates/sm120_matmul.py]: These assertions are specific to SM120 and would fail on SM100. Putting them in `compiler.py` would require a large number of `if pipeline == "sm120"` checks, making the code extremely messy.

**Review by yanqinz2 — APPROVED** [review]

**Comment by yanqinz2** [comment]
@cudnn-ci-bot run frost
